packages/flutter/lib/src/widgets/routes.dart

```diff
@@ -2440,7 +2440,24 @@ abstract class PopEntry<T> {
 
 /// Mixin for a route that can provide a delegated secondary transition to the
 /// outgoing route.
-abstract mixin class FlexibleTransitionRouteMixin<T> {
+mixin FlexibleTransitionRouteMixin<T> on TransitionRoute<T> {
   /// The delegated transition provided to the previous route.
-  late DelegatedTransitionBuilder? delegatedTransition;
+  DelegatedTransitionBuilder? get delegatedTransition;
+
+  @override
+  void didChangeNext(Route<dynamic>? nextRoute) {
+    if (nextRoute is FlexibleTransitionRouteMixin<T> && canTransitionTo(nextRoute) && navigator != null) {
+      debugPrint('Ping the navigator state');
+      navigator!.delegateTransitionBuilder = nextRoute.delegatedTransition;
```

Comment:
Is it possible to look this value up rather than saving it? Following up on the discussion in https://github.com/flutter/flutter/pull/143553#discussion_r1585494681 and https://docs.google.com/document/d/16eMliX4HsDjRmEob2lksM6JBq2_LPVM26uYGyTid1ec/edit?disco=AAABK6AVrR8.